homeassistant/components/samsungtv/entity.py

```diff
@@ -20,7 +20,7 @@ def __init__(self, *, bridge: SamsungTVBridge, config_entry: ConfigEntry) -> Non
         self._bridge = bridge
         self._mac = config_entry.data.get(CONF_MAC)
         self._attr_name = config_entry.data.get(CONF_NAME)
-        self._attr_unique_id = config_entry.unique_id
+        self._attr_unique_id = config_entry.unique_id or config_entry.entry_id
```

Comment:
Let's add a comment explaining this is a fallback for legacy models.
Please also add or update tests.